packages/kbn-esql-ast/src/types.ts

```diff
@@ -205,6 +218,20 @@ export interface ESQLStringLiteral extends ESQLAstBaseItem {
   value: string;
 }
 
+// @internal
+export interface ESQLKeywordLiteral extends ESQLAstBaseItem {
```

Comment:
We might actually not need ESQLKeywordLiteral and ESQLTextLiteral here. Since anything typed in the editor can be interpreted as a string, and get accepted as either text or keyword by the autocomplete/validation engine.